### ISSUE TITLE
Amazon Interactive Video Service (IVS): GetStream and ListStreams unable to parse ISO-8601 formatted `startTime`

### ISSUE THREAD

**Issue by kado-yasuyuki** [aws/aws-sdk-java-v2#2160]
## Describe the bug

Amazon Interactive Video Service (IVS)  [GetStream API](https://docs.aws.amazon.com/ivs/latest/APIReference/API_GetStream.html) and [ListStreams API](https://docs.aws.amazon.com/ivs/latest/APIReference/API_ListStreams.html)  throw `SdkClientException: Unable to parse date`

The [document](https://docs.aws.amazon.com/ivs/latest/APIReference/API_Stream.html#ivs-Type-Stream-startTime) describes `startTime` is `ISO-8601 formatted timestamp`.

Via the CLI, GetStream API returns the following response where `startTime` is actually in ISO-8601 format.

```json
{
    "stream": {
        "channelArn": "arn:aws:ivs:...",
        "playbackUrl": "https://....m3u8",
        "startTime": "2020-11-24T18:06:27+00:00",
        "state": "LIVE",
        "health": "HEALTHY",
        "viewerCount": 0
    }
}
```

I implemented the following simple code using this SDK.

```build.gradle:groovy
dependencies {
    implementation platform('software.amazon.awssdk:bom:2.15.34')
    implementation 'software.amazon.awssdk:ivs'
    ...
}
```

```Main.java:java
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.ivs.IvsClient;
import software.amazon.awssdk.services.ivs.model.GetStreamRequest;

public class Main {
    public static void main(String[] args) {
        var region = Region.of(args[0]);
        var channelArn = args[1];

        var ivsClient = IvsClient.builder().region(region).build();
        var req = GetStreamRequest.builder().channelArn(channelArn).build();
        var res = ivsClient.getStream(req);

        System.out.println(res);
    }
}
```

I try to get the stream of the live channel with the above code.
But it throws the following exception.

```
Exception in thread "main" software.amazon.awssdk.core.exception.SdkClientException: Unable to unmarshall response (Unable to parse date : 2020-11-24T18:06:27Z). Response Code: 200, Response Text: OK
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleSuccessResponse(CombinedResponseHandler.java:108)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleResponse(CombinedResponseHandler.java:72)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handle(CombinedResponseHandler.java:59)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handle(CombinedResponseHandler.java:40)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:40)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.HandleResponseStage.execute(HandleResponseStage.java:30)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:73)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptTimeoutTrackingStage.execute(ApiCallAttemptTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:77)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.TimeoutExceptionHandlingStage.execute(TimeoutExceptionHandlingStage.java:39)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptMetricCollectionStage.execute(ApiCallAttemptMetricCollectionStage.java:50)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallAttemptMetricCollectionStage.execute(ApiCallAttemptMetricCollectionStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:64)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:34)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:56)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.executeWithTimer(ApiCallTimeoutTrackingStage.java:80)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:60)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallTimeoutTrackingStage.execute(ApiCallTimeoutTrackingStage.java:42)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:48)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ApiCallMetricCollectionStage.execute(ApiCallMetricCollectionStage.java:31)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:37)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26)
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:193)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:133)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.doExecute(BaseSyncClientHandler.java:159)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:112)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:167)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:94)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:55)
	at software.amazon.awssdk.services.ivs.DefaultIvsClient.getStream(DefaultIvsClient.java:660)
	at Main.main(Main.java:12)
Caused by: software.amazon.awssdk.core.exception.SdkClientException: Unable to parse date : 2020-11-24T18:06:27Z
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98)
	at software.amazon.awssdk.protocols.core.StringToInstant.lambda$safeParseDate$0(StringToInstant.java:77)
	at software.amazon.awssdk.protocols.core.StringToInstant.convert(StringToInstant.java:54)
	at software.amazon.awssdk.protocols.core.StringToInstant.convert(StringToInstant.java:32)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller$SimpleTypeJsonUnmarshaller.unmarshall(JsonProtocolUnmarshaller.java:161)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller.unmarshallStructured(JsonProtocolUnmarshaller.java:213)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller.unmarshallStructured(JsonProtocolUnmarshaller.java:115)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller.unmarshallStructured(JsonProtocolUnmarshaller.java:213)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller.unmarshall(JsonProtocolUnmarshaller.java:199)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonProtocolUnmarshaller.unmarshall(JsonProtocolUnmarshaller.java:169)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonResponseHandler.handle(JsonResponseHandler.java:82)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.JsonResponseHandler.handle(JsonResponseHandler.java:36)
	at software.amazon.awssdk.protocols.json.internal.unmarshall.AwsJsonResponseHandler.handle(AwsJsonResponseHandler.java:44)
	at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.lambda$handle$0(MetricCollectingHttpResponseHandler.java:52)
	at software.amazon.awssdk.core.internal.util.MetricUtils.measureDurationUnsafe(MetricUtils.java:64)
	at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.handle(MetricCollectingHttpResponseHandler.java:52)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler$Crc32ValidationResponseHandler.handle(AwsSyncClientHandler.java:94)
	at software.amazon.awssdk.core.internal.handler.BaseClientHandler.lambda$resultTransformationResponseHandler$7(BaseClientHandler.java:269)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleSuccessResponse(CombinedResponseHandler.java:97)
	... 36 more
Caused by: java.lang.NumberFormatException: Character array is missing "e" notation exponential mark.
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:577)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:401)
	at java.base/java.math.BigDecimal.<init>(BigDecimal.java:834)
	at software.amazon.awssdk.utils.DateUtils.parseUnixTimestampInstant(DateUtils.java:135)
	at software.amazon.awssdk.protocols.core.StringToInstant.lambda$safeParseDate$0(StringToInstant.java:72)
	... 53 more
```

ListStreams API also throws a similar exception.

## Steps to Reproduce

1. create a channel in IVS
1. start video streaming to the channel using like OBS
1. make sure you can watch the live streaming on the AWS Console
1. confirm the GetStream API and ListStreams API successfully return values via CLI
1. call GetStream API and ListStream API via SDK and it throws exception.

## Your Environment

SDK version: `2.15.34`
JDK version:
```
openjdk version "11.0.9" 2020-10-20 LTS
OpenJDK Runtime Environment Corretto-11.0.9.11.1 (build 11.0.9+11-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.9.11.1 (build 11.0.9+11-LTS, mixed mode)
```
Operating System and version: `macOS Catalina 10.15.7`

**Comment by debora-ito** [comment]
@kado-yasuyuki thank you for reaching out. I'll work in the IVS setup to reproduce this.

**Comment by debora-ito** [comment]
@kado-yasuyuki I was able to reproduce the issue. 

In my case the `startTime` from GetStream API via CLI was in Z notation but both this and `2020-11-24T18:06:27+00:00` are valid ISO-8601 formats.

```
{
    "stream": {
        "channelArn": "arn:aws:ivs:...",
        "playbackUrl": "...",
        "startTime": "2020-12-07T19:38:20Z",
        "state": "LIVE",
        "health": "HEALTHY",
        "viewerCount": 0
    }
}
```

Confirming this as a bug, we will investigate the cause.

**Comment by kado-yasuyuki** [comment]
Thank you for reproducing the issue!

I think the cause of this problem is that startTime's `timestampFormat` is not `ISO-8601`.
Please refer to this pull request. https://github.com/aws/aws-sdk-java-v2/pull/2161
In my environment, this works well.

**Comment by debora-ito** [comment]
Hi @kado-yasuyuki thank you for the PR but we'll not be able to merge it. The json models are maintained by the service teams so the change must come from them, otherwise any change we make here will get overwritten when the service releases a new change.

I'll contact the IVS team internally and ask them to fix the model.

**Comment by kado-yasuyuki** [comment]
@debora-ito Ok, I understood.
I'm waiting for the IVS team to fix it.

**Comment by kado-yasuyuki** [comment]
@debora-ito Did you get a reply from the IVS team?

**Comment by debora-ito** [comment]
Yes! They are working on it, it's not a simple task it seems.

**Comment by debora-ito** [comment]
@kado-yasuyuki a fix was released in SDK version `2.15.78`, please try it out and let us know if you see any problems.

**Comment by kado-yasuyuki** [comment]
I also confirmed the fix of the problem in my environment.
The same fix as #2161 has been applied in this commit https://github.com/aws/aws-sdk-java-v2/commit/7f54e529c87e79d964731c127ced3d4b524481dd, so the timestampFormat issue has been resolved.
Thank you!
